### ISSUE TITLE
drag & drop: e.Data.GetDataPresent(DataFormats.StringFormat) does not return True while it should

### ISSUE THREAD

**Issue by rubend30** [dotnet/docs-desktop#1082]
In the following step of the walkthrough: [To process the dropped data](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/advanced/walkthrough-enabling-drag-and-drop-on-a-user-control?view=netframeworkdesktop-4.8#to-process-the-dropped-data) the first `if`-statement returns false when dropping the selected text "green" on the circle.

When debugging I added the following to check if the data was included in the `DataObject`:
   `         var color = e.Data.GetData(DataFormats.StringFormat);`

This does return the string "green" (type: System.String). Yet the next line returns false:
`if (e.Data.GetDataPresent(DataFormats.StringFormat))`

This was a one on one copy of the steps in the walkthrough.
The project is using the following target framework:
`<TargetFramework>net5.0-windows10.0.18362.0</TargetFramework>`

Thank you in advance for your time.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 23f605d0-6326-17d5-3087-4f6ef96ab9bb
* Version Independent ID: 10411c7d-3586-4904-6d4f-79e84f0c9b98
* Content: [Walkthrough: Enabling Drag and Drop on a User Control - WPF .NET Framework](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/advanced/walkthrough-enabling-drag-and-drop-on-a-user-control?view=netframeworkdesktop-4.8)
* Content Source: [dotnet-desktop-guide/framework/wpf/advanced/walkthrough-enabling-drag-and-drop-on-a-user-control.md](https://github.com/dotnet/docs-desktop/blob/main/dotnet-desktop-guide/framework/wpf/advanced/walkthrough-enabling-drag-and-drop-on-a-user-control.md)
* Product: **dotnet-framework**
* Technology: **dotnet-wpf**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

**Comment by adegeo** [comment]
@rubend30 Hi there! Sorry for the delay. We had a holiday yesterday and I wasn't available. I tried a simple test to just make sure it works as it should with a data object:

```csharp
DataObject data = new DataObject();
data.SetData(DataFormats.StringFormat, "green");

var color = data.GetData(DataFormats.StringFormat);
if (data.GetDataPresent(DataFormats.StringFormat))
{
    System.Diagnostics.Debug.WriteLine("data was present");
}
```
And it does see the string data. Can you share more code? You must be setting something incorrectly in the drag events, or perhaps there is a bug in the article code, but I can't spot one.

**Comment by rubend30** [comment]
Hello

When looking at the GIF in your answer I see that the class used is for `System.Windows.Forms.DataObject` but when looking at my code I have the `System.Windows.DataObject` class. (WPF vs. WinForms?)

I have added my project as an attachment for you to check.
[DragAndDrop.zip](https://github.com/dotnet/docs-desktop/files/6584366/DragAndDrop.zip)

Below is a GIF of my debugging results:

![draganddrop](https://user-images.githubusercontent.com/16037443/120478037-b85fa800-c3ac-11eb-8468-06be183f5ca9.gif)

**Comment by adegeo** [comment]
Whoops yeah! I thought it was a generic data object, not tied to a UI system. HA! I'll look at it some more :) Can you paste your code that sets the data?

**Comment by rubend30** [comment]
Hello

There is no code needed/provided in the walkthrough for this. I guess it is a build in functionality that when a user selects text from a TextBox that the DataObject gets set with the selected text.

There is a setData when the user presses the mouse on the circle itself but nothing for the TextBox

```
        protected override void OnMouseMove(MouseEventArgs e)
        {
            base.OnMouseMove(e);
            if (e.LeftButton == MouseButtonState.Pressed)
            {
                // Package the data.
                DataObject data = new DataObject();
                data.SetData(DataFormats.StringFormat, circleUI.Fill.ToString());
                data.SetData("Double", circleUI.Height);
                data.SetData("Object", this);

                // Inititate the drag-and-drop operation.
                DragDrop.DoDragDrop(this, data, DragDropEffects.Copy | DragDropEffects.Move);
            }
        }
```

But this is not even triggered (and shouldn't) when you drag the text on the circle. Can you test the project I attached in a previous message?

**Comment by adegeo** [comment]
OH, whoops. I didn't even notice the project you attached, yes, I'll check it out!

**Comment by adegeo** [comment]
I swapped the target framework to .NET 4.8 and it works as intended. This is a bug with .NET 5 :( I'll transfer this issue.

**Comment by adegeo** [comment]
Oh duhhh.. This whole time I've been thinking this is familiar. That I've run through this before, but I thought it was on winforms. Turns own it was already reported! They've said it will be fixed in 6.0. https://github.com/dotnet/wpf/issues/3965

**Comment by adegeo** [comment]
Duplicate of #160
